### ISSUE TITLE
Add the ability to ignore trailing-whitespace in comments

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1009]
This should probably be under a flag or we might not warn about trailing whitespaces at all, when they occur in a comment. 

The issue is part of #1005.

**Comment by aswinm** [comment]
Hi.. I am willing to work on this, and I have been going through the relevant source today. I know the check here has to be tweaked: https://github.com/PyCQA/pylint/blob/master/pylint/checkers/format.py#L962

But I am unable to get how the check for comment can be done at that place. Please suggest a way to do so.

**Comment by AWhetter** [comment]
To any new contributors, implementing a reasonable first pass at this will involve adding the comment check to https://github.com/PyCQA/pylint/blob/126c63db8010d23afd18cf65e4a6b45f4bf79efd/pylint/checkers/format.py#L989 and to also add the option to the `FormatChecker`.

If you can figure out how to consider the following case, that's great! If not then don't worry about it as a first version.

```
multi_line_string = """
# This line has trailing whitespace, but might not be detected.    
# An error should get raised because this is a string, not a comment.
```

**Comment by PCManticore** [comment]
After all I'm not sure this makes sense. Correctly configuring the editor should remove the trailing whitespaces and even `git diff --check` warns about them regardless if they are present in comments or not.
